source/DistanceMap.cpp

```diff
@@ -203,6 +203,10 @@ void DistanceMap::Init(const Ship *ship)
 		Edge top = edges.top();
 		edges.pop();
 		
+		// If the system is hidden, it cannot be used.
+		if(top.next->Hidden())
```

Comment:
Ships should still be able to make use of hidden systems if they’re linked to another system. I believe this completely disallows travel to hidden systems, which is not the intention of hidden.

Reply:
That's what the first version of this PR did, but according to Zitchas those systems should be unusable when they are hidden.

Reply:
No, originally this PR completely ignored the hidden tag if a system was linked. That’s different than a system which is hidden only being accessible via links.

Reply:
It might be easier to just allow all ships to pathfind through a hidden system if the player has seen it. It's not as though the player will be able to tell that NPCs are only now using a system that the player has found. That shouldn't be achieved by holding onto a reference to the player, though. Might it be easier for systems to also keep track of whether they've been seen/visited by the player? I believe that would also help to simplify #5415.

Reply:
Fair enough. However, if the player hasn't mapped a system - especially one marked as hidden - I don't think escorts should be using it. Escorts should be using the player's map and sticking with known routes, not finding more efficient secret ones.

On the flip side, if escorts are heading off through unexplored systems on their own, they should be revealing the player's map, since, well, the player is getting live updates as to their location, and those are ships that the player owns, so it makes sense that they'd get the nav info.

Reply:
Maybe? I don't know the code well enough to know if tracking "Have you, system, been seen by the player" makes more sense than asking "Have you, player, seen this system?"

That being said, tracking "found" status by system might make more sense if we're going to have more stuff that is stored that way. For instance:
- What is stored here?
- What ships are active here?
- What ships are parked here?

And we could extend that farther to perhaps tracking certain missions (like the static defending Archon missions) being moved from the player to the system (or planet). And going farther afield, perhaps this could be useful in generating some population "caps" for systems.

Ex. If a certain system has a "population" of 30 merchant ships, and the player kills a significant number of them, it might lower the population for a while, and as such there would be less merchants braving that system.